crates/assistant2/src/assistant_panel.rs

```diff
@@ -300,6 +300,16 @@ impl AssistantPanel {
     fn render_toolbar(&self, cx: &mut ViewContext<Self>) -> impl IntoElement {
         let focus_handle = self.focus_handle(cx);
 
+        let title = if self.thread.read(cx).is_empty() {
+            "New Thread".to_string()
+        } else {
+            self.thread
+                .read(cx)
+                .summary(cx)
+                .map(|summary| summary.to_string())
+                .unwrap_or_else(|| "Loading Summary…".to_string())
+        };
+
```

Comment:
Just acknowledging that having the "Load Summary..." here as part of the `unwrap` piece can feel like a hack. That's because, from what I understand, this text would appear also if the title summarization, for whatever reason, failed. In this scenario, "Loading Summary" would be a misleading placeholder text. As far as I know, we don't currently have a summarization enum of sorts from where we could pull a "loading" state. That might be a more robust way to tackle it, but I'm open to suggestions!